test/unit/testAlertStateMachine.js

```diff
@@ -32,413 +37,448 @@ describe('alertStateMachine.js unit tests:', () => {
+          this.returnMessageToOtherResponderPhoneNumbers = stateTransition.returnMessageToOtherResponderPhoneNumbers
+        })
 
-      it('should give the return message to otherResponderPhoneNumbers for STARTED --> WAITING_FOR_CATEGORY', () => {
-        const { returnMessageToOtherResponderPhoneNumbers } = this.alertStateMachine.processStateTransitionWithMessage(
-          CHATBOT_STATE.STARTED,
-          '3',
-          dummyIncidentCategoryKeys,
-          dummyIncidentCategories,
-          dummyLanguage,
-        )
+        it('should transition to WAITING_FOR_CATEGORY', () => {
+          expect(this.nextAlertState).to.equal(CHATBOT_STATE.WAITING_FOR_CATEGORY)
+        })
 
-        expect(returnMessageToOtherResponderPhoneNumbers).to.equal(
-          `To OtherResponderPhoneNumbers (${dummyLanguage}): ${CHATBOT_STATE.STARTED} --> ${CHATBOT_STATE.WAITING_FOR_CATEGORY} with "undefined"`,
-        )
-      })
-    })
+        it('should not change the incident category', () => {
+          expect(this.incidentCategoryKey).to.be.undefined
+        })
 
-    describe('given alert state is WAITING_FOR_REPLY', () => {
-      it('should transition to WAITING_FOR_CATEGORY', () => {
-        const { nextAlertState } = this.alertStateMachine.processStateTransitionWithMessage(
-          CHATBOT_STATE.WAITING_FOR_REPLY,
-          '3',
-          dummyIncidentCategoryKeys,
-          dummyIncidentCategories,
-          dummyLanguage,
-        )
+        it('should give the return message to respondedByPhoneNumber for STARTED --> WAITING_FOR_CATEGORY', () => {
+          expect(this.returnMessageToRespondedByPhoneNumber).to.equal(
+            dummyGetReturnMessageToRespondedByPhoneNumber(
+              dummyLanguage,
+              CHATBOT_STATE.STARTED,
+              CHATBOT_STATE.WAITING_FOR_CATEGORY,
+              dummyIncidentCategories,
+            ),
+          )
+        })
 
-        expect(nextAlertState).to.equal(CHATBOT_STATE.WAITING_FOR_CATEGORY)
+        it('should give the return message to otherResponderPhoneNumbers for STARTED --> WAITING_FOR_CATEGORY', () => {
+          expect(this.returnMessageToOtherResponderPhoneNumbers).to.equal(
+            dummyGetReturnMessageToOtherResponderPhoneNumbers(dummyLanguage, CHATBOT_STATE.STARTED, CHATBOT_STATE.WAITING_FOR_CATEGORY, undefined),
+          )
+        })
       })
 
-      it('should not change the incident category', () => {
-        const { incidentCategoryKey } = this.alertStateMachine.processStateTransitionWithMessage(
-          CHATBOT_STATE.WAITING_FOR_REPLY,
-          '3',
-          dummyIncidentCategoryKeys,
-          dummyIncidentCategories,
-          dummyLanguage,
-        )
+      describe('and messageText is a request to reset', () => {
+        beforeEach(() => {
+          const stateTransition = this.alertStateMachine.processStateTransitionWithMessage(
+            CHATBOT_STATE.STARTED,
+            dummyGetClientMessageForRequestToReset(),
```

Comment:
I'm a little confused about why this passes.

Given the definition of `dummyGetClientMessageForRequestToReset`, I was expecting this to return `'Reset ()'`

And given that `this.alertStateMachine.processStateTransitionWithMessage` was passed `dummyLanguage`, which is defined as `my_lng`, I would have expected the reset message to be `'Reset (my_lng)'`, which means that they are not equal and this shouldn't transition to RESET. Right?

Reply:
Ah, wow. Really good catch. In the alert state machine, I hadn't passed language into getClientMessageForRequestToReset, so it checked "Reset ()" === "Reset ()". I think this worked for English because no language was assumed to be English. Oops; fixing this!